extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +143,262 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,string)=} opt_processResponse An optional
+   * function to receive any response messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(ampDoc, transportOptions, opt_processResponse) {
+    dev().assert(transportOptions['iframe'],
+      'Cross-domain frame parameters missing');
+    const frameUrl = transportOptions['iframe'];
+    this.beginUsingCrossDomainIframe_(ampDoc, frameUrl,
+      transportOptions['extraData']);
+    const frameData = Transport.crossDomainIframes_[frameUrl];
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
+      'ampAnalytics3pReady', () => {
```

Comment:
Ideally the enum declaring these message types would be in a common location shared between AMP analytics and the library to ensure they remain consistent